pkg/virtctl/guestfs/guestfs.go

```diff
@@ -0,0 +1,483 @@
+package guestfs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+	"kubevirt.io/kubevirt/pkg/virtctl/utils"
+)
+
+const (
+	defaultImageName = "libguestfs-tools"
+	defaultImage     = "quay.io/kubevirt/" + defaultImageName + ":latest"
+	// KvmDevice defines the resource as in pkg/virt-controller/services/template.go, but we don't import the package to avoid compile conflicts when the os is windows
+	KvmDevice         = "devices.kubevirt.io/kvm"
+	volume            = "volume"
+	contName          = "libguestfs"
+	diskDir           = "/disks"
+	diskPath          = "/dev/vda"
+	podNamePrefix     = "libguestfs-tools"
+	appliancePath     = "/usr/local/lib/guestfs"
+	tmpDirVolumeName  = "libguestfs-tmp-dir"
+	tmpDirPath        = "/tmp/guestfs"
+	pullPolicyDefault = corev1.PullIfNotPresent
+)
+
+var (
+	pvc         string
+	image       string
+	timeout     = 500 * time.Second
+	pullPolicy  string
+	kubevirtNs  string
+	kubevirtCRD string
+	kvm         bool
+	namespace   string
+	podName     string
+)
+
+type guestfsCommand struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+// NewGuestfsShellCommand returns a cobra.Command for starting libguestfs-tool pod and attach it to a pvc
+func NewGuestfsShellCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "guestfs",
+		Short:   "Start a shell into the libguestfs pod",
+		Long:    `Create a pod with libguestfs-tools, mount the pvc and attach a shell to it. The pvc is mounted under the /disks directory inside the pod for filesystem-based pvcs, or as /dev/vda for block-based pvcs`,
+		Example: usage(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := guestfsCommand{clientConfig: clientConfig}
+			return c.run(cmd, args)
+		},
+	}
+	cmd.PersistentFlags().StringVar(&pvc, "pvc", "", "pvc claim name")
+	cmd.MarkPersistentFlagRequired("pvc")
+	cmd.PersistentFlags().StringVar(&image, "image", "", "libguestfs-tools container image")
```

Comment:
unless i'm missing something, i don't think this `image` flag works because it apperas that the code is overwriting whatever the user provides

Reply:
I lost the check in the various version... thanks!!